engine/function/src/templates/Numeric.ftl

```diff
@@ -0,0 +1,2446 @@
+     */
+    public static double log(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.log(value);
+    }
+
+    <#list primitiveTypes as pt2>
+    <#if pt2.valueType.isNumber >
+
+    /**
+     * Returns the value of the first argument raised to the second argument.
+     *
+     * @param   a   the base.
+     * @param   b   the exponent.
+     * @return {@code a} raised to the {@code b} power.
+     */
+    public static double pow(${pt.primitive} a, ${pt2.primitive} b) {
+        if (isNull(a) || isNull(b)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.pow(a, b);
+    }
+
+    </#if>
+    </#list>
+
+    /**
+     * Returns the integer closest to the input value.
+     *
+     * @param value value.
+     * @return integer closes to the input value.
+     */
+    public static double rint(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.rint(value);
+    }
+
+    /**
+     * Returns the closest integer to the argument.  If the argument is NaN, the result is 0.  If the argument is greater
+     * than {@code Integer.MIN_VALUE}, {@code Integer.MIN_VALUE} is returned.  If the argument is less than {@code Integer.MAX_VALUE},
+     * {@code Integer.MAX_VALUE} is returned.
+     *
+     * @param value value.
+     */
+    public static long round(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_LONG;
+        }
+
+        return Math.round(value);
+    }
+
+    /**
+     * Returns the signum function.
+     *
+     * @param value value.
+     * @return signum function.
+     */
+    public static int signum(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_INT;
+        }
+
+        return Integer.signum((int)value);
+    }
+
+    /**
+     * Returns the sine.
+     *
+     * @param value value.
+     * @return sine.
+     */
+    public static double sin(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.sin(value);
+    }
+
+    /**
+     * Returns the square root.
+     *
+     * @param value value.
+     * @return square root.
+     */
+    public static double sqrt(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.sqrt(value);
+    }
+
+    /**
+     * Returns the tangent.
+     *
+     * @param value value.
+     * @return tangent.
+     */
+    public static double tan(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.tan(value);
+    }
+
+
+    <#if pt.valueType.isFloat >
+
+    /**
+     * Returns the lower bound of the bin containing the value.
+     *
+     * @param value value.
+     * @param interval bin width.
+     * @return lower bound of the bin containing the value.
```

Comment:
In the previous review I suggested that the Javadocs all explain what the mathematical formula is for calculating lowerBin/upperBin (because callers won't see the code).

If you disagree, that's fine, I'm not insisting. I'm just repeating the suggestion here in case it was overlooked.